odoo/addons/test_read_group/tests/test_groupby_week.py

```diff
@@ -62,73 +64,124 @@ def setUpClass(cls):
             {'date': '2022-06-20'},       # W25, W25, W26
         ])
 
+    def add_datetime_values(self):
+        tzinfo = timezone(self.Model._context['tz'])
+        for record in self.records:
+            local_dt = tzinfo.localize(datetime.combine(record.date, time.min))
+            record.datetime = local_dt.astimezone(UTC).replace(tzinfo=None)
+
     def test_belgium(self):
         """ fr_BE - first day of the week = Monday """
+        self.Model = self.Model.with_context(lang='fr_BE', tz='Europe/Brussels')
         self.assertEqual(
             babel.Locale.parse("fr_BE").first_week_day,
             0,
         )
         self.env['res.lang']._activate_lang('fr_BE')
-        groups = self.Model.with_context(lang='fr_BE').read_group(
+        groups = self.Model.read_group(
             [('id', 'in', self.records.ids)], fields=['date'], groupby=['date:week'])
+        date_counts_by_week = {group['date:week']: group['date_count'] for group in groups}
         self.assertDictEqual(
-            {week['date:week']: week['date_count'] for week in groups if week['date:week']},
+            date_counts_by_week,
             self.per_locale["fr_BE"],
             "Week groups not matching when the first day of the week is Monday"
         )
 
         # same test as above with week_number as aggregate
-        groups = self.Model.with_context(lang='fr_BE').read_group(
+        groups = self.Model.read_group(
             [('id', 'in', self.records.ids)], fields=['date'], groupby=['date:iso_week_number'])
         self.assertDictEqual(
-            {week['date:iso_week_number']: week['date_count'] for week in groups if week['date:iso_week_number']},
+            {week['date:iso_week_number']: week['date_count'] for week in groups},
             self.iso_weeks,
             "Week groups not matching when the first day of the week is Monday"
         )
 
+        # verify grouping on datetime is identical to grouping on date
+        self.add_datetime_values()
+        groups = self.Model.read_group(
+            [('id', 'in', self.records.ids)],
+            fields=['datetime'],
+            groupby=['datetime:week'],
+        )
+        self.assertDictEqual(
+            {group['datetime:week']: group['datetime_count'] for group in groups},
+            date_counts_by_week,
```

Comment:
IMO a test should always compare the output of some function to a constant, which is the expected output.  I find this cleaner, clearer, safer.
```suggestion
        self.assertDictEqual(
            {group['datetime:week']: group['datetime_count'] for group in groups},
            self.per_locale["fr_BE"],
```